src/raft.rs

```diff
@@ -2058,4 +2117,15 @@ impl<T: Storage> Raft<T> {
     pub fn abort_leader_transfer(&mut self) {
         self.lead_transferee = None;
     }
+
+    fn send_request_snapshot(&mut self) {
+        let mut m = Message::new();
+        m.set_msg_type(MessageType::MsgAppendResponse);
+        m.set_index(self.raft_log.committed);
+        m.set_reject(true);
```

Comment:
Why set reject?

Reply:
It needs to be rejected, so it can enter requesting snapshot state in leader's view. See more at `handle_append_response`.